azkaban-web-server/src/main/java/cloudflow/servlets/ProjectServlet.java

```diff
@@ -0,0 +1,100 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProjectServlet extends LoginAbstractAzkabanServlet {
+
+  private static final String GET_ALL_PROJECT_URI = "/projects";
+  private static final String PROJECT_ID_KEY = "projectId";
+  private static final UriTemplate GET_PROJECT_URI_TEMPLATE = new UriTemplate(
+      String.format("/projects/{%s}", PROJECT_ID_KEY));
+
+  private ProjectService projectService;
+  private ObjectMapper objectMapper;
+
+  private static final Logger log = LoggerFactory.getLogger(ProjectServlet.class);
+
+  @Override
+  public void init(final ServletConfig config) throws ServletException {
+    super.init(config);
+    final AzkabanWebServer server = (AzkabanWebServer) getApplication();
+    this.projectService = server.projectService();
+    this.objectMapper = server.objectMapper();
+  }
+
+  @Override
+  protected void handleGet(HttpServletRequest req, HttpServletResponse resp, Session session)
+      throws IOException, ServletException {
+    Map<String, String> templateVariableToValue = new HashMap<>();
+    if (GET_ALL_PROJECT_URI.equals(req.getRequestURI())) {
+      /* Get all records */
+      String response = objectMapper.writeValueAsString(projectService.getAllProjects(session.getUser()));
+      this.writeResponse(resp, response);
+      return;
+    } else if (GET_PROJECT_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
+      /* Get specific record */
+      try {
+        String projectId = templateVariableToValue.get(PROJECT_ID_KEY);
+
+        /* Validate projectId is an Integer. */
+        try {
+          Integer.parseInt(projectId);
+        } catch (NumberFormatException e) {
+          log.error("Invalid project id: ", projectId);
+          sendErrorResponse(resp, HttpServletResponse.SC_BAD_REQUEST, "Invalid project id");
+          return;
+        }
+
+        String response = objectMapper.writeValueAsString(projectService.getProject(projectId));
+        this.writeResponse(resp, response);
+      } catch (Exception e) {
+        log.error("Exception while fetching project: " + e);
+        sendErrorResponse(resp, HttpServletResponse.SC_NOT_FOUND, "Project id not found");
+        return;
+      }
+    } else {
+      /* Unsupported route, return an error */
+      log.error("Invalid route for projects endpoint: " + req.getRequestURI());
+      sendErrorResponse(resp, HttpServletResponse.SC_NOT_IMPLEMENTED, "Unsupported projects API "
+          + "endpoint");
+      return;
+    }
+  }
+
+  @Override
+  protected void handlePost(HttpServletRequest req, HttpServletResponse resp, Session session)
+      throws ServletException, IOException {
+    try {
+      String body = HttpRequestUtils.getBody(req);
+      Project project = objectMapper.readValue(body, Project.class);
+
+      String projectId = projectService.createProject(project, session.getUser());
+      sendResponse(resp, HttpServletResponse.SC_CREATED, projectId);
```

Comment:
Can you add a "Location" header in the response? Something like: `resp.setHeader("Location", "/projects/" + projectId);`
Also, in the current specification POST requests don't return any data (besides the location header).

Reply:
Setting the location now and not sending projectId in response body.